frappe/public/js/frappe/form/multi_select_dialog.js

```diff
@@ -1,110 +1,76 @@
-			},
-			{
-				fieldtype: "Column Break"
-			}
-		];
-		let count = 0;
-		if(!this.date_field) {
-			this.date_field = "transaction_date";
-		}
+		let fields = [];
 
-		// setters can be defined as a dict or a list of fields
-		// setters define the additional filters that get applied
-		// for selection
+		let header_fields = this.make_dialog_header();
 
-		// CASE 1: DocType name and fieldname is the same, example "customer" and "customer"
-		// setters define the filters applied in the modal
-		// if the fieldnames and doctypes are consistently named,
-		// pass a dict with the setter key and value, for example
-		// {customer: [customer_name]}
-
-		// CASE 2: if the fieldname of the target is different,
-		// then pass a list of fields with appropriate fieldname
+		fields = fields.concat(header_fields);
 
-		if($.isArray(this.setters)) {
-			for (let df of this.setters) {
-				fields.push(df, {fieldtype: "Column Break"});
-			}
-		} else {
-			Object.keys(this.setters).forEach(function(setter) {
-				fields.push({
-					fieldtype: me.target.fields_dict[setter].df.fieldtype,
-					label: me.target.fields_dict[setter].df.label,
-					fieldname: setter,
-					options: me.target.fields_dict[setter].df.options,
-					default: me.setters[setter]
-				});
-				if (count++ < Object.keys(me.setters).length) {
-					fields.push({fieldtype: "Column Break"});
-				}
-			});
+		if (this.add_filters_group) {
+			fields = fields.concat([
+				{
+					fieldtype: 'HTML',
+					fieldname: 'filter_area',
+				},
+			]);
 		}
 
 		fields = fields.concat([
-			{
-				"fieldname":"date_range",
-				"label": __("Date Range"),
-				"fieldtype": "DateRange",
-			},
-			{ fieldtype: "Section Break" },
 			{ fieldtype: "HTML", fieldname: "results_area" },
-			{ fieldtype: "Button", fieldname: "more_btn", label: __("More"),
-				click: function(){
+			{
+				fieldtype: "Button", fieldname: "more_btn", label: __("More"),
+				click: function () {
 					me.start += 20;
 					frappe.flags.auto_scroll = true;
 					me.get_results();
 				}
 			}
 		]);
 
+		if (this.data_fields) {
+			fields = fields.concat([{ fieldtype: "Section Break" }]);
+			fields = fields.concat(this.data_fields);
+
+		}
+
 		let doctype_plural = !this.doctype.endsWith('y') ? this.doctype + 's'
 			: this.doctype.slice(0, -1) + 'ies';
 		this.dialog = new frappe.ui.Dialog({
-			title: __("Select {0}", [(this.doctype=='[Select]') ? __("value") : __(doctype_plural)]),
+			title: __("Select {0}", [(this.doctype == '[Select]') ? __("value") : __(doctype_plural)]),
 			fields: fields,
-			primary_action_label: __("Get Items"),
+			primary_action_label: this.primary_action_label || __("Get Items"),
 			secondary_action_label: __("Make {0}", [me.doctype]),
-			primary_action: function() {
-				me.action(me.get_checked_values(), me.args);
+			primary_action: function () {
+				let filters_data = me.get_filters();
+				me.action(me.get_checked_values(), cur_dialog.get_values(), me.args, filters_data);
```

Comment:
Why are we passing filters data?

Reply:
Because sometimes it is necessary to show filters.
for example
![image](https://user-images.githubusercontent.com/32095923/81178358-b2d73900-8fc5-11ea-88fc-c390ba374b7b.png)
As filters need to be shown in payroll entry for employees

Reply:
Is this not passed in args already? The arguments here are not very clean.